course/src/main/java/org/openedx/course/presentation/ui/CourseVideosUI.kt

```diff
@@ -125,19 +124,21 @@ fun CourseVideosScreen(
                 )
             }
         },
-        onDownloadClick = {
-            if (viewModel.isBlockDownloading(it.id)) {
-                viewModel.courseRouter.navigateToDownloadQueue(
-                    fm = fragmentManager,
-                    viewModel.getDownloadableChildren(it.id)
-                        ?: arrayListOf()
-                )
-            } else if (viewModel.isBlockDownloaded(it.id)) {
-                viewModel.removeDownloadModels(it.id)
-            } else {
-                viewModel.saveDownloadModels(
-                    FileUtil(context).getExternalAppDir().path, it.id
-                )
+        onDownloadClick = { blocksIds ->
+            blocksIds.forEach { blockId ->
+                if (viewModel.isBlockDownloading(blockId)) {
+                    viewModel.courseRouter.navigateToDownloadQueue(
+                        fm = fragmentManager,
+                        viewModel.getDownloadableChildren(blockId)
+                            ?: arrayListOf()
+                    )
```

Comment:
can we transform it to the one line `viewModel.getDownloadableChildren(blockId) ?: arrayListOf()` for better readability?